modules/system/classes/VersionManager.php

```diff
@@ -115,6 +115,33 @@ public function replacePlugin(PluginBase $plugin, string $replace)
             return;
         }
 
+        $code = $plugin->getPluginIdentifier();
+
+        // add history up to $currentVersion
+        if ($versions = $this->getOldFileVersions($code, $currentVersion)) {
+            foreach ($versions as $version => $details) {
+                list($comments, $scripts) = $this->extractScriptsAndComments($details);
+
+                foreach ($scripts as $script) {
+                    Db::table('system_plugin_history')->insert([
+                        'code'       => $code,
+                        'type'       => self::HISTORY_TYPE_SCRIPT,
+                        'version'    => $version,
+                        'detail'     => $script,
+                        'created_at' => new Carbon
+                    ]);
+                }
+
+                foreach ($comments as $comment) {
+                    $this->applyDatabaseComment($code, $version, $comment);
+                }
+            }
+        }
+
+        // delete replaced plugin history
+        Db::table('system_plugin_history')->where('code', $replace)->delete();
+
+        // replace installed version
         Db::table('system_plugin_versions')
             ->where('code', '=', $replace)
             ->update([
```

Comment:
Can't leave a comment where I wanted to, but the line below (line: 148) can use `$code` as well rather than calling the `getPluginIdentifier()` function again

Reply:
yes, thanks!